### ISSUE TITLE
scale_y_log10: hidden segments

### ISSUE THREAD

**Issue by ASmirnov-HORIS** [JetBrains/lets-plot#372]
Look at the following example:

```python
from lets_plot import *
LetsPlot.setup_html()
n = 10
x = list(range(n))
y = [2**d for d in range(n)]
ggplot({'x': x, 'y': y}) + \
    geom_segment(aes(x='x', y=[0]*n, xend='x', yend='y')) + \
    geom_point(aes('x', 'y')) + \
    scale_y_log10()
```

It produces an image without segments:

<img src="https://user-images.githubusercontent.com/26162403/118799251-2bdec100-b8a7-11eb-9756-0b91a2749d0c.png" width="400" height="300" />

Segments appear if we remove `scale_y_log10()` layer:

<img src="https://user-images.githubusercontent.com/26162403/118799601-93950c00-b8a7-11eb-8f34-39135a37716e.png" width="400" height="300" />

**Comment by alshan** [comment]
I suspect we loose segments because their lower Y (=0) becomes -Infinity on log10 scale.

**Comment by alshan** [comment]
Later I reverted the 'fix': excluded 0.0 from log10 domain. So, the code here will produce plot without segments again(.
The workaround would be:
a) define segment 'y' as constant: `geom_segment(aes(x='x', xend='x', yend='y'), y=0)`
b) make 'y'-s not 0 but just small: `geom_segment(aes(x='x', y=[1e-3]*n, xend='x', yend='y'))`
